frontend/src/data/Dialect.ts

```diff
@@ -32,14 +31,70 @@ export default class Dialect {
   }
 
   async fetchReport(baseURI: URI) {
-    const url = baseURI.clone().filename(this.path).suffix("json").href();
-    const response = await fetch(url);
+    const curVersionUrl = baseURI
+      .clone()
+      .filename(this.path)
+      .suffix("json")
+      .href();
+    const response = await fetch(curVersionUrl);
     const jsonl = await response.text();
-    const lines = jsonl
+    const curReportLines = jsonl
       .trim()
       .split(/\r?\n/)
       .map((line) => JSON.parse(line) as Record<string, unknown>);
-    return parseReportData(lines);
+
+    let curReport = parseReportData(curReportLines);
+
+    const prevVersionUrl = baseURI
```

Comment:
It strikes me as odd that a function meant for retrieving a report is also retrieving the previous report. That seems like something that should be happening outside of here.

You seem to be doing it because of the new added `isNew` property -- but maybe that's indeed worth *not* putting on `Implementation`, it doesn't seem like it does much except create the possibility to be wrong.

Why not just have this function be more or less as it was, then have a *separate* one which pulls the `RunInfo` from yesterday and gives you a `Set` of yesterday's implementations? Then your `isNew` is just checking for set membership.

Reply:
So if I am understanding it correctly you are saying that the isNew Field is not needed and instead we could pass that data as a prop maybe ?

Reply:
Yeah basically you can calculate it when it's needed from the combination of one implementation and a list of yesterdays -- so if you pass that list of yesterday's implementations to the page which wants to highlight,.you can highlight all those implementations not appearing in the yesterday list.

Does that make sense?

Reply:
Yeahh, got it